nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/FXMLController.java

```diff
@@ -603,30 +603,28 @@ public boolean isFIDActive() {
 
     void addFID(NMRData nmrData, boolean clearOps, boolean reload) {
         isFID = true;
-        if (chartProcessor == null) {
-            if (processorController == null) {
-                processorController = ProcessorController.create(this, processorPane, getActiveChart());
-            }
+        if (chartProcessor == null && processorController == null) {
+            // The create method also creates and sets the chartProcessor variable
+            processorController = ProcessorController.create(this, processorPane, getActiveChart());
         }
-        chartProcessor.setData(nmrData, clearOps);
         if (processorController != null) {
+            chartProcessor.setData(nmrData, clearOps);
             processorController.viewingDataset(false);
             processorController.updateFileButton();
             processorController.show();
+            processorController.clearOperationList();
+            chartProcessor.clearAllOperations();
+            processorController.parseScript("");
+            if (!reload) {
+                getActiveChart().full();
+                getActiveChart().autoScale();
+                chartProcessor.loadDefaultScriptIfPresent();
+            }
+            getActiveChart().layoutPlotChildren();
+            statusBar.setMode(0);
```

Comment:
moved possible NPE code into the null check and added a log message, not sure if I should do anything else here? the previous behaviour would have just not added the FID because of the NPE without any message to the user, should there be a dialog to the user?
Also no null check needed for chartProcessor because if ProcessorController.create creates the processorController then chartProcessor will not be null.

Reply:
I don't think these will ever be null in normal usage so no message is needed to user.  I think the null checks and pritnlns  were put in during some initial testing where there was a problem in loading fxml files.